source/server/config_validation/async_client.cc

```diff
@@ -3,7 +3,8 @@
 namespace Envoy {
 namespace Http {
 
-ValidationAsyncClient::ValidationAsyncClient(Api::Api& api) : dispatcher_(api) {}
+ValidationAsyncClient::ValidationAsyncClient(Api::Api& api, Event::TimeSystem& time_system)
+    : dispatcher_(api, time_system) {}
```

Comment:
I'm probably forgetting something but why does this class allocate a dispatcher? That seems odd.

Reply:
I reasoned this out with the help of the compiler. ValidationAsyncClient implements pure interface AsyncClient. AsyncClient has a dispatcher() method. If I make that method private in the abstract base class, I get compiler errors in source/common/grpc/async_client_impl.cc which needs to save a dispatcher as a member variable, in order to later perform a deferred delete.
```
    dispatcher_->deferredDelete(
        LinkedObject<AsyncStreamImpl>::removeFromList(parent_.active_streams_));
```

There were no other dependencies in //source/... that I could find, though there may be some in //test/... That doesn't seem like a super-strong reason to have to carry this dispatcher, but that seems lie a refactor for another day.

Reply:
Got it, sounds good.